python/python/glide/async_commands/server_modules/ft_options/ft_aggregate_options.py

```diff
@@ -0,0 +1,322 @@
+
+
+class Filter(FtAggregateClause):
+    """
+    A clause for filtering the results using predicate expression relating to values in each result. It is applied post query and relate to the current state of the pipeline.
+    """
+
+    def __init__(self, expression: TEncodable):
+        """
+        Initialize a new Filter instance.
+
+        Args:
+            expression (TEncodable): The expression to filter the results.
+        """
+        self.expression = expression
+
+    def toArgs(self) -> List[TEncodable]:
+        """
+        Get the arguments for the Filter clause.
+
+        Returns:
+            List[TEncodable]: A list arguments for the filter clause.
+        """
+        args = super().toArgs()
+        args.extend([FtAggregateKeywords.FILTER, self.expression])
+        return args
+
+
+class Reducer:
+    """
+    A clause for reducing the matching results in each group using a reduction function. The matching results are reduced into a single record.
+    """
+
+    def __init__(
+        self,
+        function: TEncodable,
+        args: List[TEncodable],
+        name: Optional[TEncodable] = None,
+    ):
+        """
+        Initialize a new Reducer instance.
+
+        Args:
+            function (TEncodable): The reduction function names for the respective group.
+            args (List[TEncodable]): The list of arguments for the reducer.
+            name (Optional[TEncodable]): User defined property name for the reducer.
+        """
+        self.function = function
+        self.args = args
+        self.name = name
+
+    def toArgs(self) -> List[TEncodable]:
+        """
+        Get the arguments for the Reducer.
+
+        Returns:
+            List[TEncodable]: A list of arguments for the reducer.
+        """
+        args: List[TEncodable] = [FtAggregateKeywords.REDUCE, self.function]
+        args.append(str(len(self.args)))
+        if self.args:
+            args.extend(self.args)
+        if self.name:
+            args.extend([FtAggregateKeywords.AS, self.name])
+        return args
+
+
+class GroupBy(FtAggregateClause):
+    """
+    A clause for grouping the results in the pipeline based on one or more properties.
+    """
+
+    def __init__(
+        self, properties: List[TEncodable], reducers: Optional[List[Reducer]] = None
+    ):
+        """
+        Initialize a new GroupBy instance.
+
+        Args:
+            properties (List[TEncodable]): The list of properties to be used for grouping the results in the pipeline.
+            reducers (Optional[List[Reducer]]): The list of functions that handles the group entries by performing multiple aggregate operations.
+        """
+        self.properties = properties
+        self.reducers = reducers
+
+    def toArgs(self) -> List[TEncodable]:
+        """
+        Get the arguments for the GroupBy clause.
+
+        Returns:
+            List[TEncodable]: A list arguments for GroupBy clause.
+        """
+
+        args = super().toArgs()
```

Comment:
super args are always empty - why we need that?

Reply:
Updated.